pkg/metrics/metrics.go

```diff
@@ -187,6 +192,29 @@ func (pc *PrometheusNodeConfig) Configure() error {
 	return nil
 }
 
+// createMetricsEndpoint creates an endpoint object in the operator namespace.
+// WMCO is no longer creating a service with selector hence no Endpoint object is created
+// and WMCO needs to create it.
```

Comment:
I don't follow this statement. This too is being created by WMCO.

Reply:
The emphasis here is on `selector`. Previously the `service` had selector field by default  that was responsible for creating the `Endpoints` object and we were removing in the later step.

Since we can now create our own resource definition, we create a service without `selector`.  See [Service without selector](https://kubernetes.io/docs/concepts/services-networking/service/#services-without-selectors) for reference.

Reply:
Please clarify this in the comment. Saying `it` makes it unclear which object you are referring to.